datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolverTest.java

```diff
@@ -120,4 +137,80 @@ public void testGetSuccessNoParentNode() throws Exception {
         Mockito.any(Authentication.class)
     );
   }
+
+  @Test
+  public void testGetFailureExistingTermSameName() throws Exception {
+    EntityClient mockClient = Mockito.mock(EntityClient.class);
+
+    Mockito.when(
+        mockClient.filter(
+            Mockito.eq(Constants.GLOSSARY_TERM_ENTITY_NAME),
+            Mockito.any(),
+            Mockito.eq(null),
+            Mockito.eq(0),
+            Mockito.eq(1000),
+            Mockito.any()
+        )
+    ).thenReturn(new SearchResult().setEntities(
+        new SearchEntityArray(new SearchEntity().setEntity(UrnUtils.getUrn(EXISTING_TERM_URN)))
+    ));
+
+    Map<Urn, EntityResponse> result = new HashMap<>();
+    EnvelopedAspectMap map = new EnvelopedAspectMap();
+    GlossaryTermInfo termInfo = new GlossaryTermInfo().setName("Duplicated Name");
+    map.put(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME, new EnvelopedAspect().setValue(new Aspect(termInfo.data())));
+    result.put(UrnUtils.getUrn(EXISTING_TERM_URN), new EntityResponse().setAspects(map));
+
+    Mockito.when(
+        mockClient.batchGetV2(
+            Mockito.eq(Constants.GLOSSARY_TERM_ENTITY_NAME),
+            Mockito.any(),
+            Mockito.eq(Collections.singleton(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME)),
+            Mockito.any()
+        )
+    ).thenReturn(result);
+
+    EntityService mockService = Mockito.mock(EntityService.class);
+    DataFetchingEnvironment mockEnv = Mockito.mock(DataFetchingEnvironment.class);
+
+    CreateGlossaryEntityInput input = new CreateGlossaryEntityInput(
+        "test-id",
+        "Duplicated Name",
+        "test-description",
+        "urn:li:glossaryNode:12372c2ec7754c308993202dc44f548b"
+    );
+    setupTest(mockEnv, input, "test-description", parentNodeUrn);
+    CreateGlossaryTermResolver resolver = new CreateGlossaryTermResolver(mockClient, mockService);
+    assertThrows(CompletionException.class, () -> resolver.get(mockEnv).join());
+
+    Mockito.verify(mockClient, Mockito.times(0)).ingestProposal(
+        Mockito.any(),
+        Mockito.any(Authentication.class)
+    );
+  }
+
+  private EntityClient initMockClient() throws Exception {
```

Comment:
You could probably repurpose this for the testGetFailureExistingTermSameName if you support passing in a SearchResult and a result Hashmap

Reply:
oh you right